scheduler/src/test/java/org/cloudfoundry/autoscaler/scheduler/util/TestDataDbUtil.java

```diff
@@ -190,4 +199,25 @@ private int convertArrayToBits(int[] values) {
 		return bits;
 	}
 
+	public enum DatabaseType {
+		POSTGRESQL, MYSQL,
+	}
+
+	public DatabaseType getDatabaseTypeFromDataSource() throws Exception {
+		if (this.databaseType != null) {
+			return this.databaseType;
+		}
+		String driverName = this.dataSource.getConnection().getMetaData().getDriverName().toLowerCase();
+		if (driverName.indexOf("postgresql") > -1) {
```

Comment:
check  driverName not null and not empty before doing any further operation.

Reply:
Added the check for dirverName.

Reply:
@cdlliuy @aqan213 
If we need to do empty/null check, so we need to check dataSource, getConnection, getMetaData, getDriverName to ensure the return value is not null.